docs(stack): reorganize navigation into evaluate and deployment paths (#1605)
Consensys/doc.linea: docs/stack/how-it-works/deployment-models.mdx

---
title: Deployment models
description: >-
  How to compare Lineth public and private validium deployment models and their
  trade-offs
sidebar_position: 2
image: /img/socialCards/deployment-models.jpg
---

Use this page to compare documented Lineth deployment models and the trade-offs
they create for access, data availability, and finalization. Related evaluation
paths include [trust and responsibilities](./trust-model.mdx) and

> [privacy and data visibility](../features/validium.mdx).

Reviewer: ```suggestion
This page describes the Lineth deployment models and the trade-offs
they create for access, data availability, and finalization. Related evaluation
paths include [Trust and responsibilities](./trust-model.mdx) and
[Privacy and data visibility](../features/validium.mdx).
```

"Use this page to..." on every page feels a little awkward, suggest switching to "This page [describes/compares X]" for concept pages.

## Customization factors

> Lineth is designed to support customizations for factors including:

Reviewer: ```suggestion
An operator can customize the following in a Lineth deployment:
```

- Access control: who may access the mempool, ledger, and transaction data
- Data availability: how transaction data is stored and who can reconstruct it
- Finalization speed: block size and configuration affect soft finality speed

> - Finalization layer: where finalization occurs affects security assumptions and

Reviewer: ```suggestion
- Finalization layer: whether you finalize on Ethereum or Linea affects security assumptions and
```

  hard finality speed
- Transaction ordering: operators can apply custom ordering rulesets

Operators own these choices. Lineth does not prescribe a single deployment
shape, and the public docs do not assert that any specific configuration is

> production-ready or commercially supported.

Reviewer: ```suggestion
Operators own these choices. Lineth does not recommend a specific set of customizations.
```

It's kind of odd to self-refer to the "public docs."

## Two primary models

Lineth supports two primary deployment models:

- Public deployments, such as Linea Mainnet, with onchain data availability

> - Private validium deployments, with offchain data availability and controlled

Reviewer: ```suggestion
- [Public deployments](#public), such as Linea Mainnet, with onchain data availability
- [Private validium deployments](#private-validium), with offchain data availability and controlled
```

  access

When configuring a deployment, consider:

- Privacy requirements: should transaction data remain private to authorized
  participants?
- Regulatory and compliance obligations: what controls does your program

>   require, and who owns them? See [Compliance](../features/compliance.mdx).

Reviewer: ```suggestion
- Regulatory and [compliance](../features/compliance.mdx) obligations: what controls does your program
  require, and who owns them?
```

### Use cases

- Workflows where transaction data should not be public
- Multi-party workflows where participants should not see all transaction
  details
- Operators who accept the validium trade-off: state-transition correctness is
  proven onchain, but data reconstruction without the operator depends on the
  deployment's data availability arrangement

Operators with regulatory or compliance obligations are responsible for
designing their own compliance program. See
[Compliance](../features/compliance.mdx) for what protocol evidence exists and
what is not claimed.

### Security considerations

- Validator topology: at least 4 Maru validators are required for a
  [QBFT design](../how-to/distributed-sequencing.mdx) that tolerates one faulty
  validator
- Access controls: [RBAC](../features/rbac.mdx) on RPC endpoints and API portal
- [Key management](../../protocol/architecture/index.mdx#web3signer): remote
  signing backed by a hardware security module (HSM) or key management service
  (KMS)
- Network isolation: private network topology with controlled peering

### Example validium system setup

The following network setup presents how an operator-run validium instance of

> Lineth may be configured. The operator determines where transaction data is

Reviewer: ```suggestion
The following is an example validium network setup.
The operator determines where transaction data is
```
